### PR TITLE
 #27959-search: adding Embeddings data generation convenience class and SearchToolTest as an integration test

### LINKED ISSUE #27959: Integration tests for AI Viewtools
### Parent Issue

https://github.com/dotCMS/core/issues/27835

### Task

- Write automated tests for all viewtools
    - $ai 
        - [x] $ai.generateText
        - [x] $ai.generateImage
    - $ai.search
        - [x] $ai.search.query
        - [x] $ai.search.related
    - $ai.embeddings
        - [x] $ai.embeddings.countTokens
        - [x] $ai.embeddings.generateEmbeddings
        - [x] $ai.embeddings.getIndexCount
        - [x] $ai.embeddings.indexCount
    - $ai.completions
        - [x] $ai.completions.getConfig
        - [x] $ai.completions.config
        - [x] $ai.completions.summarize
        - [x] $ai.completions.raw

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### Acceptance Criteria

- [ ] All the viewtools working according to the documentation https://www.dotcms.com/docs/latest/dotai-plugin#Viewtool
- [ ] Test coverage for all viewtools

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsAPI.java [hunk 1]
```diff
@@ -40,7 +47,7 @@ static EmbeddingsAPI impl() {
      * @param index
      * @return
      */
-    boolean generateEmbeddingsforContent(Contentlet contentlet, List<Field> fields, String index);
+    boolean generateEmbeddingsForContent(Contentlet contentlet, List<Field> fields, String index);
```
Comment (freddyDOTCMS) [inline dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsAPI.java:50]: doc

File: dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsAPI.java [hunk 2]
```diff
@@ -51,7 +58,7 @@ static EmbeddingsAPI impl() {
      * @param indexName
      * @return
      */
-    boolean generateEmbeddingsforContent(@NotNull Contentlet contentlet, String velocityTemplate, String indexName);
+    boolean generateEmbeddingsForContent(@NotNull Contentlet contentlet, String velocityTemplate, String indexName);
```
Comment (freddyDOTCMS) [inline dotCMS/src/main/java/com/dotcms/ai/api/EmbeddingsAPI.java:61]: doc?

**Review by freddyDOTCMS — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/ai/api/BulkEmbeddingsRunner.java [hunk 3]
```diff
@@ -37,8 +37,8 @@ public void run() {
                 List<Field> fields = contentlet.getContentType().fields().stream().filter(f -> embeddingsForm.fieldsAsList().contains(f.variable().toLowerCase())).collect(Collectors.toList());
 
                 // if a velocity template is passed in, use it.  Otherwise, try the fields
-                if (!EmbeddingsAPI.impl().generateEmbeddingsforContent(contentlet, embeddingsForm.velocityTemplate, embeddingsForm.indexName)) {
-                    EmbeddingsAPI.impl().generateEmbeddingsforContent(contentlet, fields, embeddingsForm.indexName);
+                if (!EmbeddingsAPI.impl().generateEmbeddingsForContent(contentlet, embeddingsForm.velocityTemplate, embeddingsForm.indexName)) {
+                    EmbeddingsAPI.impl().generateEmbeddingsForContent(contentlet, fields, embeddingsForm.indexName);
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/api/BulkEmbeddingsRunner.java:41]: Let's make sure the runners call API methods only and that we have the proper transaction annotations for these API methods.

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/ai/api/BulkEmbeddingsRunner.java (hunk 3 above)
Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/ai/api/BulkEmbeddingsRunner.java:41]: Will be done by @vicozizou in the next PR

**Pull request merged**